paramiko/proxy.py

```diff
@@ -81,6 +81,11 @@ def recv(self, size):
 
         :return: the string of bytes read, which may be shorter than requested
         """
+        if self.process.poll() is not None:
+            # Cleanup zombie.
+            self.process.wait()
```

Comment:
`.wait()` is not really needed here - if `.poll()` did not return None, it already did the wait on the child and got the returncode and returned it. poll() is just like using libc wait() with WNOHANG